splash/jsutils.py

```diff
@@ -76,6 +96,45 @@ def get_sanitized_result_js(expression, max_depth=0):
     )
 
 
+STORE_DOM_ELEMENTS_JS = u"""
+function (elements_storage_name, o) {
+  var storage = window[elements_storage_name];
+
+  function storeNode(node) {
+    var id = storage.getId();
+
+    Object.defineProperty(storage, id, {
+      configurable: false,
+      enumerable: false,
+      writable: false,
+      value: node,
+    });
+
+    return {
+      type: 'node',
+      id: id
+    };
+  }
+
+  if (o instanceof Node) {
+    return storeNode(o);
+  } else if (o instanceof NodeList) {
+    return Array.prototype.slice.call(o).map(storeNode);
+  }
+
+  return o;
```

Comment:
Does it mean user can return `{type: 'node', id: '...'}` and Splash will think a DOM element was returned? What do you think about always wrapping the result here - for non-node objects it can return e.g. `{type: 'any', value: ...}, and for node lists it can return `{type: 'nodelist', 'value': ...}`, and the unwrapping the result?